MIP/mip-1/README.md

```diff
@@ -0,0 +1,113 @@
+- **APP**: the Client sends an application message to the Enclave, a 256-bit nonce for the current message, and a 256-bit nonce for the next message.
+- **APP-OK**: the Enclave responds indicating it has processed the application message and has updated the nonce for the next message.
+- **APP-OK-CON**: the Enclave responds indicating that it has processed the application message, has updated the nonce for the next message, and has aborted a contentious synchronization attempt. 
+- **APP-REJ**: the Enclave responds indicating it has received the application message but is rejecting it and will not update the nonce for the next message.
+
+These messages are exchanged as follows:
+
+**Synchronization Phase**
+1. Prior to sending its first application message, the Client generates and safely stores CSPRNG nonce and sends it to the Enclave in a **SYN** message.
+2. The Enclave processes the **SYN** message:
+    - If enclave does not have a stored nonce, this indicates that no other client has attempted to synchronize with the Enclave. The Enclave MUST accept the nonce, store the nonce, and respond with a **SYN-OK** message.
+    - If the Enclave has a record of the nonce, this indicates that another client has attempted to synchronize with the Enclave in the past. It MUST check queue and time-lock conditions:
+        - If the nonce is at the top of the queue and has passed its time-lock, the Enclave MUST accept, store, and respond with a **SYN-OK** message.
+        - If the nonce is not at the top of the queue or has not passed its time-lock, the Enclave MUST respond with a **SYN-TL** message.
+    
+**Application Phase**
+1. The Client sends an **APP** message to the Enclave with the application message and the nonce for the current message and the nonce for the next message.
+2. The Enclave processes the **APP** message:
+    - If the nonce for the current message is the expected nonce, the Enclave MUST process the application message and update the nonce for the next message. It MUST responds with an **APP-OK** message.
+    - If the nonce for the current message is not the expected nonce, the Enclave MUST respond with an **APP-REJ** message.
+    - If the nonce for the current message is the expected nonce but the Enclave has nonces in its time-lock queue, i.e., from other SYN messages, the Enclave MUST:
+        1. Empty the queue of all other nonces;
+        2. Process the application message;
+        3. Respond with an **APP-OK-CON** message.
+3. The Client MUST wait for an **APP-OK**, **APP-REJ**, or **APP-OK-CON** message before sending the next application message.
+    - If the message is **APP-OK** or **APP-OK-CON**, the Client MUST send the next application message with the "next nonce" specified in the previous message.
+    - If the message is **APP-REJ**, the Client MUST re-send the application message with the "current nonce" it used in the previous message.
```

Comment:
Will this cause an endless loop if the Enclave works as specified above and rejects the current nonce with APP-REJ?